libs/common/service.go

```diff
@@ -124,6 +131,8 @@ func (bs *BaseService) Start() error {
 	if atomic.CompareAndSwapUint32(&bs.started, 0, 1) {
 		if atomic.LoadUint32(&bs.stopped) == 1 {
 			bs.Logger.Error(fmt.Sprintf("Not starting %v -- already stopped", bs.name), "impl", bs.impl)
+			// revert flag
+			atomic.StoreUint32(&bs.started, 0)
```

Comment:
I'm confused as to why we have separate started and stopped flags. Shouldn't `started = !stopped`?

Reply:
There was a discussion about simplifying `Service` and @jaekwon made a PR https://github.com/tendermint/tmlibs/pull/114. Not sure why we did not merge it.